src/Microsoft.ServiceFabric.Client.Http/ImageStoreClient.cs

```diff
@@ -34,128 +35,193 @@ public ImageStoreClient(ServiceFabricHttpClient httpClient)
+                return await this.httpClient.SendAsyncGetResponse(RequestFunc, url, ImageStoreContentConverter.Deserialize, requestId, cancellationToken);
+            }
         }
 
         /// <inheritdoc />
-        public Task DeleteImageStoreContentAsync(
+        public async Task DeleteImageStoreContentAsync(
             string contentPath,
             long? serverTimeout = 60,
             CancellationToken cancellationToken = default(CancellationToken))
         {
             contentPath.ThrowIfNull(nameof(contentPath));
-            serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
-            var requestId = Guid.NewGuid().ToString();
-            var url = "ImageStore/{contentPath}";
-            url = url.Replace("{contentPath}", Uri.EscapeDataString(contentPath));
-            var queryParams = new List<string>();
-            
-            // Append to queryParams if not null.
-            serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
-            queryParams.Add("api-version=6.0");
-            url += "?" + string.Join("&", queryParams);
-            
-            HttpRequestMessage RequestFunc()
+
+            await LoadImageStoreConnectionString();
+            if (isLocalStore)
             {
-                var request = new HttpRequestMessage()
-                {
-                    Method = HttpMethod.Delete,
-                };
-                return request;
+                var path = Path.Combine(imageStorePath, contentPath);
+                FileAttributes attr = File.GetAttributes(path);
+
+                if ((attr & FileAttributes.Directory) == FileAttributes.Directory)
+                    Directory.Delete(path, true);
+                else
+                    File.Delete(path);
             }
+            else
+            {
+                serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
+                var requestId = Guid.NewGuid().ToString();
+                var url = "ImageStore/{contentPath}";
+                url = url.Replace("{contentPath}", Uri.EscapeDataString(contentPath));
+                var queryParams = new List<string>();
 
-            return this.httpClient.SendAsync(RequestFunc, url, requestId, cancellationToken);
+                // Append to queryParams if not null.
+                serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
+                queryParams.Add("api-version=6.0");
+                url += "?" + string.Join("&", queryParams);
+
+                HttpRequestMessage RequestFunc()
+                {
+                    var request = new HttpRequestMessage()
+                    {
+                        Method = HttpMethod.Delete,
+                    };
+                    return request;
+                }
+
+                await this.httpClient.SendAsync(RequestFunc, url, requestId, cancellationToken);
```

Comment:
This can just be return, 

[https://stackoverflow.com/questions/21033150/any-difference-between-await-task-run-return-and-return-task-run](https://stackoverflow.com/questions/21033150/any-difference-between-await-task-run-return-and-return-task-run)

Reply:
It can not, since it's an async method and it has to be async since it retrieves the connection string.